src/DataTable/TableCol.js

```diff
@@ -63,14 +63,14 @@ const TableCol = memo(({
       let direction = sortDirection;
       // change sort direction only if sortColumn (currently selected column) is === the newly clicked column
       // otherwise, retain sort direction if the column is switched
-      if (sortColumn === column.selector) {
+      if (sortColumn === column.id) {
         direction = sortDirection === 'asc' ? 'desc' : 'asc';
       }
 
       dispatch({
         type: 'SORT_CHANGE',
         sortDirection: direction,
-        sortColumn: column.selector,
+        sortColumn: column.id,
```

Comment:
Looks like this breaks sorting  because `sortColumn` is expected to be string based on the selector name.

Reply:
`sortColumn` probably needs to be converted to the currently selected column object. It should probably also be renamed to be more clear

Reply:
Right. I didn't test with built-in sorting, only server sorting. This action does have the entire column object in its payload.

I'd be a bit wary of major refactoring in a plain JS project. How interested would you be in switching to TS? Or at least introducing TS type checking into the code (I think it's possible to add it to JS code through comments)

Reply:
I've had plans to convert the entire lib to TS - I just haven't had the time to dedicate to it.  Along with this would be a major refactor and possible breaking changes.

Reply:
I think for now it's fine in the JS code as long as it's commented. As I mentioned it will eventually be TS as soon as I either get some cycles or get some help 😉

Reply:
I don't mind helping you out with that. I'm looking to get more involved in open source anyway :) I work in TS regularly and your lib already has typings, that facilitates it.

But we can go with comments for now too. I haven't worked with them all that much, but it's not complicated. Would just get bulky, that's all.